### PR TITLE
[17.09] Load dataset attribute editing and errors without page refresh

### LINKED ISSUE #4601: minor regressions in the updated dataset 'pencil' menu
as of https://github.com/galaxyproject/galaxy/commit/26c7abb130bade73b64284956f0077f35067095f

* dataset’s ‘edit attributes’ on dev branch refresh the page before displaying the form
* the history box highlight (so you can see what dataset are you operating on) is not present anymore

### PR DESCRIPTION
**Pull request by guerler**
Addition to #4334. Partially fixes #4601.

### CODE REVIEW

**Comment by nsoranzo** [comment]
The failed `py27-lint` Travis build can be safely ignored.

@jmchilton It seems we need to backport https://github.com/galaxyproject/galaxy/pull/4766 or pin also `pydocstyle` in tox.ini, because flake8-docstrings simply [requires](https://gitlab.com/pycqa/flake8-docstrings/blob/master/setup.py#L47) `'pydocstyle >= 2.0'` .

File: client/galaxy/scripts/mvc/dataset/dataset-li-edit.js [hunk 1]
```diff
@@ -49,20 +49,24 @@ var DatasetListItemEdit = _super.extend(
...
             deleted = this.model.get( 'deleted' ),
             editBtnData = {
                 title       : _l( 'Edit attributes' ),
                 href        : Galaxy.root + 'datasets/edit?dataset_id=' + this.model.attributes.id,
                 faIcon      : 'fa-pencil',
-                classes     : 'edit-btn'
+                classes     : 'edit-btn',
+                onclick     : function( ev ) {
+                    ev.preventDefault();
+                    Galaxy.router.push( 'datasets/edit', { dataset_id : self.model.attributes.id } );
```
Comment (martenson) [inline client/galaxy/scripts/mvc/dataset/dataset-li-edit.js]: this will not work on /view_multiple since there is no router I think

Comment (guerler) [reply client/galaxy/scripts/mvc/dataset/dataset-li-edit.js]: Thanks for catching that.

**Comment by martenson** [comment]
Thanks @guerler !

**Pull request merged**